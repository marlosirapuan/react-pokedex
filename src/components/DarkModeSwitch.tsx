import React from 'react'

import { Box, Switch, useColorMode } from '@chakra-ui/react'
import { FiMoon, FiSun } from 'react-icons/fi'

type DarkModeSwitchProps = {
  position: 'relative' | 'absolute'
}

const DarkModeSwitch = ({ position }: DarkModeSwitchProps): JSX.Element => {
  const { colorMode, toggleColorMode } = useColorMode()

  return (
    <Box
      display="flex"
      flexDirection="row"
      alignItems="center"
      position={position}
    >
      {colorMode === 'dark' ? <FiMoon /> : <FiSun />}

      <Switch
        title={`${colorMode === 'light' ? 'Dark' : 'Light'} mode`}
        size="md"
        aria-label="toggle dark mode"
        marginLeft={2}
        isChecked={colorMode === 'dark'}
        onChange={toggleColorMode}
      />
    </Box>
  )
}

export default DarkModeSwitch
