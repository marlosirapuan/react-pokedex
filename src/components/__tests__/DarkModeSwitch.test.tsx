import React from 'react'
import { render } from '@testing-library/react'

import DarkModeSwitch from '../DarkModeSwitch'

test('renders the component', () => {
  const { getByLabelText } = render(<DarkModeSwitch position="relative" />)

  const darkModeSwitchElement = getByLabelText(/toggle dark mode/i)

  expect(darkModeSwitchElement).toBeInTheDocument()
})
