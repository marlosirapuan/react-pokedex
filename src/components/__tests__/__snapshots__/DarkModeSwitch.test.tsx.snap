// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DarkModeSwitch component should be able to render 1`] = `ShallowWrapper {}`;
